src/data/datasets/representation/validation.py

```diff
@@ -0,0 +1,288 @@
+from cmath import pi
+from pathlib import Path
+from typing import Tuple, Union, Optional
+
+import torch
+import pandas as pd
+from albumentations import BasicTransform
+from albumentations.core.composition import BaseCompose
+
+from src.data.datasets.base import ImageDataset
+
+
+class RetrievalDataset(ImageDataset):
+    """Dataset for query->relevant benchmarking
+    Dataset for image retrieval where gallery consists of query, relevant and gallery items(optional).
+    The searches are made by queries while looking for relevant items in the whole set of items.
+    Where gallery items are treated non-relevant.
+
+    Example matches csv:
+    Query ids should be unique, otherwise the rows having the same query id will be treated as different matches.
+    Relevant ids can be repeated in different queries.
+    Scores reflect the order of similarity of the image to the query,
+    a higher score corresponds to a greater similarity.
+
+    .. csv-table:: Match csv example
+    :header: query, relevant, scores
+    1194917,601566 554492 224125 2001716519,4 3 2 2
+    1257924,456490,4
+
+    Example img_list csv:
+    img_list.csv maps the id's of query and relevant elements to image paths
+
+    .. csv-table:: Image csv example
+    :header: id, image_path
+    1194917,data/img_1.jpg
+    601566,data/img_2.jpg
+    554492,data/img_3.jpg
+    224125,data/img_4.jpg
+    2001716519,data/img_5.jpg
+    1257924,data/img_6.jpg
+    456490,data/img_7.jpg
+
+    .. csv-table:: Gallery Image csv example
+    :header: id, image_paths
+    8,data/db/img_1.jpg
+    10,data/db/img_2.jpg
+    12,data/db/img_3.jpg
+    """
+
+    def __init__(self,
+                 data_folder: str,
+                 matches_csv_path: str,
+                 img_list_csv_path: str,
+                 transform: Union[BasicTransform, BaseCompose],
+                 augment: Optional[Union[BasicTransform, BaseCompose]] = None,
+                 use_gallery: Optional[bool] = False,
```

Comment:
You can remove this argument - when a user specifies `gallery_folder` we can automatically derive that this is a gallery mode